sdk/python/kfp/components/_python_op.py

```diff
@@ -328,15 +328,31 @@ def annotation_to_type_struct(annotation):
     input_names = set()
     output_names = set()
     for parameter in parameters:
-        parameter_annotation = parameter.annotation
+        parameter_type = parameter.annotation
```

Comment:
I wonder what is the main motivation for this change?
Not all annotations are types. In fact most KFP annotations are not types, but are strings (type names).